GenerateOriginalFiles.py

```diff
@@ -0,0 +1,234 @@
+
+# Add all the modules and the destinations to the dictionary.
+Dictionary['FreeRTOS_ARP.c'] = ARP_modules_to_combine
+Dictionary['FreeRTOS_DHCP.c'] = DHCP_modules_to_combine
+Dictionary['FreeRTOS_DNS.c'] = DNS_modules_to_combine
+Dictionary['FreeRTOS_IP.c'] = IP_modules_to_combine
+Dictionary['FreeRTOS_Sockets.c'] = Socket_modules_to_combine
+Dictionary['FreeRTOS_Stream_Buffer.c'] = StreamBuffer_modules_to_combine
+Dictionary['FreeRTOS_TCP_IP.c'] = TCP_modules_to_combine
+Dictionary['FreeRTOS_TCP_WIN.c'] = TCP_WIN_modules_to_combine
+Dictionary['FreeRTOS_UDP_IP.c'] = UDP_modules_to_combine
+
+# Sorting function used to add Kernel includes. This is required as the includes
+# need to adhere to a specific order of inclusion.
+def KernelSortingFunction(element):
+    return FreeRTOS_Kernel_Includes.index(element)
+
+# Sorting function used to add FreeRTOS+TCP includes. This is required as the includes
+# need to adhere to a specific order of inclusion.
+def TCPSortingFunction(element):
+    return FreeRTOS_TCP_Includes.index(element)
+
+# Get all the includes in all the modules to combine first. This function does
+# not remove duplicates neither does it sort the includes in any specific order.
+def GetIncludeList( ModulesToCombine ):
+    # This list is used for the FreeRTOS+TCP include files.
+    TCP_include_list = list()
+    # This list is used for FreeRTOS Kernel include files.
+    Kernel_include_list = list()
+    # This list is used for standard library include files.
+    StdLib_include_list = list()
+
+    for filename in ModulesToCombine:
+        f = open(filename, "r")
+        for line in f.readlines():
+            if line.lstrip().startswith('#include '):
+                if ( '<' in line ) and ( line not in StdLib_include_list ):
+                    #if this is a standard library include
+                    start_token = '<'
+                    end_token = '>'
+                    header = line.split(start_token)[1].split(end_token)[0]
+                    StdLib_include_list.append( header)
+                else:
+                    start_token = '"'
+                    end_token = '"'
+                    header = line.split(start_token)[1]
+
+                    if header in FreeRTOS_TCP_Includes:
+                        TCP_include_list.append(header)
+                    elif header in FreeRTOS_Kernel_Includes:
+                        Kernel_include_list.append(header)
+                    else:
+                        print( "ERROR: Found " + header + " which is not in any list!" )
+        f.close()
+    return StdLib_include_list, Kernel_include_list, TCP_include_list
+
+# Add the includes in a specific order to the destination file.
+def AddIncludesInFile( modules_to_combine, file_descriptor ):
+    StdLib_include_list, Kernel_include_list, TCP_include_list = GetIncludeList( modules_to_combine )
+
+    StdLib_include_list_unique = list( set( StdLib_include_list ) )
```

Comment:
This seems like an extra step when the previous function could just return a `set` to begin with, and/or do the `set` -> `list` conversion in the `return` if it's also meant to be sorted later